### PR TITLE
Fix failed fetch request not unblocking completion

### PR DESCRIPTION
**Pull request by Sushisource**
## What was changed
If fetching pages failed while in the middle of an activation completion, that could leave the completion stuck because the eviction wouldn't unblock it. This fixes that.

## Why?
Bug

## Checklist

1. Closes 

2. How was this tested:

3. Any docs updates needed?

### CODE REVIEW

File: core/src/worker/workflow/history_update.rs [hunk 1]
```diff
@@ -147,12 +152,13 @@ impl HistoryPaginator {
             npt,
             client,
         );
+        if empty_hist && wft.legacy_query.is_none() && wft.query_requests.is_empty() {
+            return Err(EMPTY_TASK_ERR.clone());
```
Comment (mjameswh) [inline core/src/worker/workflow/history_update.rs:156]: Q: Does lang needs to expect and catch this error?

Comment (Sushisource) [reply core/src/worker/workflow/history_update.rs:156]: Nope. You don't need to do anything special, it's just going to show up in evictions / logs

File: core/src/worker/workflow/mod.rs [hunk 2]
```diff
@@ -829,6 +829,7 @@ struct RequestEvictMsg {
     run_id: String,
     message: String,
     reason: EvictionReason,
+    because_fetch_failed: bool,
```
Comment (mjameswh) [inline core/src/worker/workflow/mod.rs:832]: Could that special purpose field have been avoided? I don't know if adding a new `EvictionReason` would be better?

Comment (Sushisource) [reply core/src/worker/workflow/mod.rs:832]: Good point, I'll change that

**Review by Spikhalskiy — APPROVED** [review]

**Pull request merged**